evalml/pipelines/utils.py

```diff
@@ -253,26 +253,30 @@ def generate_pipeline_code(element):
     return "\n".join(code_strings)
 
 
-def _make_stacked_ensemble_pipeline(input_pipelines, problem_type, n_jobs=-1, random_state=0):
+def _make_stacked_ensemble_pipeline(input_pipelines, problem_type, data_splitter=None, n_jobs=-1, random_state=0):
     """
     Creates a pipeline with a stacked ensemble estimator.
 
     Arguments:
         input_pipelines (list(PipelineBase or subclass obj)): List of pipeline instances to use as the base estimators for the stacked ensemble.
             This must not be None or an empty list or else EnsembleMissingPipelinesError will be raised.
-        problem_type (ProblemType): problem type of pipeline
+        data_splitter (sklearn.model_selection.BaseCrossValidator): Data splitting method for the stacking ensemble to use.
+            If set to None, will use the default stacking ensemble data split. Defaults to None.
+        problem_type (ProblemType): Problem type of pipeline
         n_jobs (int or None): Integer describing level of parallelism used for pipelines.
             None and 1 are equivalent. If set to -1, all CPUs are used. For n_jobs below -1, (n_cpus + 1 + n_jobs) are used.
             Defaults to -1.
 
     Returns:
         Pipeline with appropriate stacked ensemble estimator.
     """
+    if data_splitter is not None and data_splitter.get_n_splits() < 2:
```

Comment:
This is going to raise `ValueError` in `AutoMLSearch` when the dataset is large or the user manually passes in a `TrainingValidationSplit`.That would stop the search early and show a stacktrace so I think we should do something about it in `AutoMLSearch`. I think there are two things we can do:

1. I think we should set `ensembling=False` under-the-hood in `AutoMLSearch` and log a message if the user passes in a data splitter with not enough splits.
2. Raise a value error in init if the data splitter does not have enough splits.

I think I prefer 1, that's kind of what we do when users don't specify enough batches or pipelines, but what do you think?

Reply:
I guess the deciding point is: either AutoML smartly handle the user's desired data splitting strategy and then does an AutoML search (that's expensive in time?) or stops early the moment it knows the data split was unsuccessful.  Let me know if I've characterized that correctly. 

If we raise a value error in init, is there necessarily anything a user can do about the number of splits?  At that point would they have to increase the size of their dataset?  I kind of feel that if I'm personally going to use an AutoML search algorithm, I'd like to know before I kick it off that I won't benefit from a StackedEnsemble being in the running rather than get to the end to find that my dataset wasn't large enough.